### PR TITLE
Sentiment nba twitter

### LINKED ISSUE #10: Create SentimentAnalyzer
The SentimentAnalyzer will take in a pandas dataframe generated by the SentimentModeler. The Sentiment Modeler will get this information from one or many NBAPlayer objects.

The SentimentAnalyzer should then append the sentiment values to the last column of the incoming df. The modified df will then be sent back to the SentimentModeler.

### PR DESCRIPTION
**Pull request by afandino**
My small contribution, use pip install --upgrade azure-ai-textanalytics

### CODE REVIEW

File: src/main/python/SentimentAnalyzer.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import pandas as pd
+# pip install --upgrade azure-ai-textanalytics
+from azure.ai.textanalytics import TextAnalyticsClient
+from azure.core.credentials import AzureKeyCredential
+
+_key = "adc96f39ac2a4918a1f462b5bf8674b8"
```
Comment (darkhark) [inline src/main/python/SentimentAnalyzer.py:6]: It's usually not a good idea to hardcode api keys into code. We can either make this repo private and give the professor access when she needs to grade or there's a few other options (keyring, system property, etc.). If you don't mind this key being public, we could also just leave it as is. I just don't want to leave a security vulnerability out there in for someone outside this group to use your key.

File: src/main/python/SentimentAnalyzer.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+_endpoint = "https://nbatwittersentiment.cognitiveservices.azure.com/"
+
+def _authenticate(key:str,endpoint:str):
```
Comment (darkhark) [inline src/main/python/SentimentAnalyzer.py]: Private methods use two underscores. I also thought it was one when I first started this project lol.

File: src/main/python/SentimentAnalyzer.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    # Authentication
+    ta_credential = AzureKeyCredential(key)
+    client = TextAnalyticsClient(
+        endpoint=endpoint,
+        credential=ta_credential,
+        api_version='TextAnalyticsApiVersion.V3_0')
+    return client
+
+def _formatresponse(responseDict) -> pd.DataFrame :
```
Comment (darkhark) [inline src/main/python/SentimentAnalyzer.py]: Can you add some docs for at least what the responseDict is? I think that's the dict of tweetID, tweet, but I'm not sure. Also, given our conversation earlier today, that dict may become date, tweet instead since the date would be a unique identifier.
Overall it's a bit hard to understand this method. It looks like responseDict may not be a dictionary but some other sort of object?

File: src/main/python/SentimentAnalyzer.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    col_list = responseDict.keys()
+    cleanresponse = {'tweetID':responseDict.id,
+                    'positive':responseDict.confidence_scores.positive,
+                    'neutral':responseDict.confidence_scores.neutral,
+                    'negative':responseDict.confidence_scores.negative
+                    }
+    for idx, sentence in enumerate(responseDict.sentences):
+        cleanresponse ['tweet']= sentence.text
+    # sentiment_df = pd.DataFrame.from_dict(cleanresponse,orient='index')
+    sentiment_df = pd.DataFrame.from_records(cleanresponse,index=[0])
```
Comment (darkhark) [inline src/main/python/SentimentAnalyzer.py:32]: What does this do?

File: src/main/python/SentimentAnalyzer.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    return sentiment_df
+
+def sentiment_analysis(tweet) -> pd.DataFrame:
+    """Uses Microsoft Cognitive Service Text Analytics 3.0 to analyze text
+    @param tweet[] list: a list of dictionary tweet objects
```
Comment (darkhark) [inline src/main/python/SentimentAnalyzer.py]: This is a bit confusing. So the incoming variable "tweet" is a list of dictionaries where the dictionaries are {id, tweet}? Why would it need to be a list of dictionaries?

File: src/main/python/SentimentAnalyzer.py (hunk 3 above)
Comment (darkhark) [reply src/main/python/SentimentAnalyzer.py]: Jk, date can't be used as a key because of mixing different players together into the same df/dictionary. The key will stay tweetID.

File: src/main/python/NBADataLoader.py [hunk 6]
```diff
@@ -1,29 +0,0 @@
-from nba_api.stats.static import players
```
Comment (darkhark) [inline src/main/python/NBADataLoader.py]: Why was this file deleted? I think you may need to merge in the latest main. This file should not exist in main already due to a name and location change.

**Review by darkhark — CHANGES_REQUESTED** [review]
Just a few things to look over before I'll approve. I think your branch might be pretty far behind the main branch. If you need help updating your main, I think there's a message in the group chat that is pined that should give you the git commands to update your local main branch.

**Comment by darkhark** [comment]
Also, did you have any test code with this? I've just been leaving the test code in as a comment at the end that way if changes are needed in the future, we can make sure we didn't break anything. Normally I would write these as unit tests but I don't want to overwhelm everyone lol.

File: src/main/python/SentimentAnalyzer.py (hunk 1 above)
Comment (afandino) [reply src/main/python/SentimentAnalyzer.py:6]: Agree, I will remove the key.

File: src/main/python/SentimentAnalyzer.py (hunk 2 above)
Comment (afandino) [reply src/main/python/SentimentAnalyzer.py]: private methods were refactored

File: src/main/python/SentimentAnalyzer.py (hunk 3 above)
Comment (afandino) [reply src/main/python/SentimentAnalyzer.py]: Added doc from Microsoft, it is a list of objects.

File: src/main/python/SentimentAnalyzer.py (hunk 4 above)
Comment (afandino) [reply src/main/python/SentimentAnalyzer.py:32]: removed the commented line. I was not sure what orientation we needed, I matched the output to the other sentiment analyzer from @el_Grudge. Since this client can take in multiple or single documents, I needed to step through the list to pull the text. This should work for one or multiple tweets submitted. I also pulled the overall scores for the request. If more than one tweet is submitted the overall sentiment is returned.

File: src/main/python/SentimentAnalyzer.py (hunk 5 above)
Comment (afandino) [reply src/main/python/SentimentAnalyzer.py]: this was poorly worded and laziness. The incoming variable can just be a dictionary that should include the tweeID, I wanted to preserve the functionality to submit multiple tweets at once. I can do this inside the method.

File: src/main/python/NBADataLoader.py (hunk 6 above)
Comment (afandino) [reply src/main/python/NBADataLoader.py]: yes forgot to merge in the latest main it was late.

File: src/main/python/SentimentAnalyzer.py (hunk 4 above)
Comment (darkhark) [reply src/main/python/SentimentAnalyzer.py:32]: I more meant what the from_records does lol. I can just look that up though. Figured I'd ask in case you had a quick thing you could say.

File: src/main/python/SentimentAnalyzer.py (hunk 1 above)
Comment (darkhark) [reply src/main/python/SentimentAnalyzer.py:6]: What was your plan for removing this?

File: src/main/python/SentimentAnalyzer.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        client = __authenticate(_key, _endpoint)
+        response = client.analyze_sentiment(documents=tweets)[0]
+        tweetdf = __formatresponse(response)
+    else:
+        print('tweetID is not present')
+        tweetdf = pd.DataFrame()
+    return tweetdf
+
+# Test code
+tweet = {
```
Comment (darkhark) [inline src/main/python/SentimentAnalyzer.py:52]: I'll test this after I'm done with my changes. For now though, can you keep this commented out? Just that way it doesn't run during normal execution.

**Review by darkhark — CHANGES_REQUESTED** [review]
Overall looks good! Just a couple little things left.
